include/sheath_boundary_parallel.hxx

```diff
@@ -0,0 +1,115 @@
+///
+/// These are based on
+/// "Boundary conditions for the multi-ion magnetized plasma-wall transition"
+///  by D.Tskhakaya, S.Kuhn. JNM 337-339 (2005), 405-409
+///
+/// Notes:
+///   - The approximation used here is for ions having similar
+///     gyro-orbit sizes
+///   - No boundary condition is applied to neutral species
+///   - Boundary conditions are applied to field-aligned fields
+///     using to/fromFieldAligned
+///
+struct SheathBoundaryParallel : public Component {
+  /// # Input options
+  /// - <name>  e.g. "sheath_boundary"
+  ///   - wall_potential           Voltage of the wall [Volts]
+  ///   - floor_potential          Apply floor to sheath potential?
+  ///   - secondary_electron_coef  Effective secondary electron emission coefficient
+  ///   - sin_alpha                Sine of the angle between magnetic field line and wall surface (0 to 1)
+  ///   - always_set_phi           Always set phi field? Default is to only modify if already set
+  SheathBoundaryParallel(std::string name, Options &options, Solver *);
+
+  ///
+  /// # Inputs
+  /// - species
+  ///   - e
+  ///     - density
+  ///     - temperature
+  ///     - pressure    Optional
+  ///     - velocity    Optional
+  ///     - mass        Optional
+  ///     - adiabatic   Optional. Ratio of specific heats, default 5/3.
+  ///   - <ions>  if charge is set (i.e. not neutrals)
+  ///     - charge
+  ///     - mass
+  ///     - density
+  ///     - temperature
+  ///     - pressure     Optional
+  ///     - velocity     Optional. Default 0
+  ///     - momentum     Optional. Default mass * density * velocity
+  ///     - adiabatic    Optional. Ratio of specific heats, default 5/3.
+  /// - fields
+  ///   - phi    Optional. If not set, calculated at boundary (see note below)
+  ///
+  /// # Outputs
+  /// - species
+  ///   - e
+  ///     - density      Sets boundary
+  ///     - temperature  Sets boundary
+  ///     - velocity     Sets boundary
+  ///     - energy_source
+  ///   - <ions>
+  ///     - density      Sets boundary
+  ///     - temperature  Sets boundary
+  ///     - velocity     Sets boundary
+  ///     - momentum     Sets boundary
+  ///     - energy_source
+  /// - fields
+  ///   - phi   Sets boundary
+  ///
+  /// If the field phi is set, then this is used in the boundary condition.
+  /// If not set, phi at the boundary is calculated and stored in the state.
+  /// Note that phi in the domain will not be set, so will be invalid data.
+  ///
+  ///
+  void transform(Options &state) override;
+private:
+  BoutReal Ge; // Secondary electron emission coefficient
+  BoutReal sin_alpha; // sin of angle between magnetic field and wall.
+
+  bool always_set_phi; ///< Set phi field?
+
+  Field3D wall_potential; ///< Voltage at the wall. Normalised units.
+
+  bool floor_potential; ///< Apply floor to sheath potential?
+
+  bool legacy_match{true};
+
+  Field3D fromFieldAligned(const Field3D& f) {
```

Comment:
I think these to/fromFieldAligned functions should be renamed: The caller may expect that the result is transformed, but if using FCI then it is not. That affects whether the caller can treat `f(x, y+1, z)` as being along a field line from `f(x, y, z)`.

I'm coming to the opinion that trying to handle both yup/down and field-aligned cases with the same code is going to be fragile and easy to introduce errors. I think I would prefer a Mesh- or ParallelTransform-level boolean: Are we using yup/down fields for differencing or not?